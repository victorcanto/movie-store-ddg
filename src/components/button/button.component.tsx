import * as React from 'react';
import { ButtonProps, Button as MuiButton, useTheme } from '@mui/material';

export function Button({
	children,
	onClick,
	disabled,
	component = '' as React.ElementType,
	to = '/',
	...restProps
}: ButtonProps & { component?: React.ElementType; to?: string }) {
	const theme = useTheme();
	return (
		<MuiButton
			data-testid='button'
			fullWidth
			component={component}
			to={to}
			variant='contained'
			onClick={onClick}
			disabled={disabled}
			{...restProps}
		>
			{children}
		</MuiButton>
	);
}
