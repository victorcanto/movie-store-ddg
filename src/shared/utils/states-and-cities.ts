import statesAndCities from '../data/estados-cidades.json';

export const getBrazilianStates = () =>
	statesAndCities.estados.map(({ nome }) => nome);

export const getBrazilianCitiesByState = (stateName: string) =>
	statesAndCities.estados.find(({ nome }) => nome === stateName)?.cidades;
