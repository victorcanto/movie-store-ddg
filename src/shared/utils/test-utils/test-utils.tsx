import React, { PropsWithChildren } from 'react';
import { render as rtlRender } from '@testing-library/react';
import type { RenderOptions } from '@testing-library/react';
import type { PreloadedState } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import { setupStore, AppStore, RootState } from '../../../store';

interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
	preloadedState?: PreloadedState<RootState>;
	store?: AppStore;
}

export function renderWithProviders(
	ui: React.ReactElement,
	{
		preloadedState = {},
		store = setupStore(preloadedState),
		...renderOptions
	}: ExtendedRenderOptions = {},
) {
	function Wrapper({ children }: PropsWithChildren<{}>): JSX.Element {
		return (
			<Provider store={store}>
				<BrowserRouter>{children}</BrowserRouter>
			</Provider>
		);
	}
	return { store, ...rtlRender(ui, { wrapper: Wrapper, ...renderOptions }) };
}

export * from '@testing-library/react';

export { renderWithProviders as render };
