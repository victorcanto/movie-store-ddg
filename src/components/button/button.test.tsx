import React from 'react';
import { render } from '../../shared/utils/test-utils';
import { Button } from './button.component';

test('snapshot of the base state', () => {
	const { asFragment } = render(<Button />);
	expect(asFragment()).toMatchSnapshot();
});

test('should have in the document', () => {
	const { getByTestId } = render(<Button />);
	expect(getByTestId('button')).toBeInTheDocument();
});
