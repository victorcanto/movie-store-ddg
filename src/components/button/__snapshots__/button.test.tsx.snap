// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot of the base state 1`] = `
<DocumentFragment>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth css-1fu7jd5-MuiButtonBase-root-MuiButton-root"
    data-testid="button"
    tabindex="0"
    to="/"
  >
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;
